runtime/tr.source/trj9/optimizer/InlinerTempForJ9.cpp

```diff
@@ -349,57 +349,71 @@ TR_J9InlinerPolicy::mustBeInlinedEvenInDebug(TR_ResolvedMethod * calleeMethod, T
    }
 
+/* Identify methods for which the benefits of inlining them into the caller
+   are particularly significant.
+*/
 bool
 TR_J9InlinerPolicy::alwaysWorthInlining(TR_ResolvedMethod * calleeMethod, TR::Node *callNode)
    {
-   if (calleeMethod && isInlineableJNI(calleeMethod, callNode))
+   // if we are passed a null calleeMethod then we can only return false
+   if (!calleeMethod) return false;
+
+   if (isInlineableJNI(calleeMethod, callNode))
       return true;
 
-   if (calleeMethod && calleeMethod->isDAAWrapperMethod())
+   if (calleeMethod->isDAAWrapperMethod())
       return true;
 
-   if (calleeMethod)
+   switch (calleeMethod->convertToMethod()->getMandatoryRecognizedMethod())
       {
-      switch (calleeMethod->convertToMethod()->getMandatoryRecognizedMethod())
-         {
-         case TR::java_lang_invoke_MethodHandle_asType:
-         case TR::java_lang_invoke_MethodHandle_invokeExactTargetAddress:
-            return true;
-         default:
-          break;
-         }
-
-      // we rely on inlining compareAndSwap so we see the inner native call and can special case it
-      // get/putLongVolatile is not supported by all codegens so we want to inline to expose the native
-      // if we haven't special cased it in the walker
-      switch (calleeMethod->getRecognizedMethod())
-         {
-         case TR::java_lang_J9VMInternals_fastIdentityHashCode:
-         case TR::java_lang_Class_getSuperclass:
-         case TR::java_lang_String_regionMatches:
-         case TR::java_lang_Class_newInstance:
-         case TR::com_ibm_jit_JITHelpers_compareAndSwapIntInObject:
-         case TR::com_ibm_jit_JITHelpers_compareAndSwapLongInObject:
-         case TR::com_ibm_jit_JITHelpers_compareAndSwapObjectInObject:
-         case TR::com_ibm_jit_JITHelpers_compareAndSwapIntInArray:
-         case TR::com_ibm_jit_JITHelpers_compareAndSwapLongInArray:
-         case TR::com_ibm_jit_JITHelpers_compareAndSwapObjectInArray:
-         case TR::com_ibm_jit_JITHelpers_jitHelpers:
-         case TR::java_lang_String_charAtInternal_I:
-         case TR::java_lang_String_charAtInternal_IB:
-         case TR::java_lang_String_length:
-         case TR::java_lang_String_lengthInternal:
-         case TR::java_lang_String_isCompressed:
-         case TR::java_lang_StringBuffer_capacityInternal:
-         case TR::java_lang_StringBuffer_lengthInternalUnsynchronized:
-         case TR::java_lang_StringBuilder_capacityInternal:
-         case TR::java_lang_StringBuilder_lengthInternal:
-         case TR::java_util_HashMap_get:
-         case TR::java_util_HashMap_getNode:
-            return true;
-         default:
-          break;
-         }
+      case TR::java_lang_invoke_MethodHandle_asType:
+      case TR::java_lang_invoke_MethodHandle_invokeExactTargetAddress:
+         return true;
+      default:
+       break;
+      }
+
+   // we rely on inlining compareAndSwap so we see the inner native call and can special case it
```

Comment:
redundant - the comment belongs with the specific list of enum entries that have the special handling. so please move this down there / incorporate into the other ocmments you added there